OpenLiberty/guide-microprofile-config
# Qa to Master: dev mode

Converted guide to run with dev mode. The follow changes were made: 

- use the ideal(simplified) pom.xml

- update the Readme content to use dev mode

  - keep the flow same (let discuss if the flow not work)

- convert *Test class to *IT
- convert to use JUnit 5
  - use JUnit5 test order if need

- make sure to correctly configure the ports in test classes, server.xml, and pom.xml

- update travis test
  - mvn install cannot be used to verify anymore
